src/lib.rs

```diff
@@ -83,18 +96,46 @@ impl<'a, 'b> DrawingBackend for RatatuiBackend<'a, 'b> {
         coord1: BackendCoord,
         coord2: BackendCoord,
         style: &S,
-        _fill: bool,
+        fill: bool,
     ) -> std::result::Result<(), DrawingErrorKind<Self::ErrorType>> {
-        let (x1, y1) = backend_to_canvas_coords(coord1, self.size);
-        let (x2, y2) = backend_to_canvas_coords(coord2, self.size);
-
-        self.canvas.draw(&canvas::Rectangle {
-            x:      x1.min(x2),
-            y:      y1.min(y2),
-            width:  (x2 - x1).abs(),
-            height: (y2 - y1).abs(),
-            color:  convert_color(style.color()),
-        });
+        if fill {
+            let (mut start, mut stop) =
+                if coord1.0 < coord2.0 { (coord1, coord2) } else { (coord2, coord1) };
+
+            let color = convert_color(style.color());
+
+            log::debug!("");
+            while start.0 <= stop.0 && start.1 <= stop.1 {
```

Comment:
Why not just use draw line by line? Why do we have to make it so complicated and reuse `canvas::Rectangle`?

Also you can use `RangeFull.zip(RangeFull)` to avoid mutability.